### PR TITLE
KBC-1611 retry password reset

### PR DESCRIPTION
**Pull request by zajca**
Jira: KBC-1611

Before asking for review make sure that:

- [x] New client method(s) has tests
- [x] Apiary file is updated
- [x] You declared if there is a BC break or not (will affect next release of a client)

---

### CODE REVIEW

File: tests/Backend/WorkspaceCredentialsAssertTrait.php [hunk 1]
```diff
@@ -2,17 +2,28 @@
...
+        $retryPolicy = new CallableRetryPolicy(function (\Exception $e) {
+            return $e->getMessage() === self::$RETRY_FAIL_MESSAGE;
+        });
         try {
-            $this->getDbConnection($connection);
-            $this->fail('Credentials should be invalid');
+            $proxy = new RetryProxy($retryPolicy, new ExponentialBackOffPolicy());
+            $proxy->call(function () use ($connection) {
+                $this->getDbConnection($connection);
+                throw new \Exception(self::$RETRY_FAIL_MESSAGE);
```
Comment (zajca) [inline tests/Backend/WorkspaceCredentialsAssertTrait.php:25]: místo $this->fail to hodí exception když dojdou retry

File: tests/Backend/WorkspaceCredentialsAssertTrait.php [hunk 2]
```diff
@@ -2,17 +2,28 @@
 
 namespace Keboola\Test\Backend;
 
+use Retry\BackOff\ExponentialBackOffPolicy;
+use Retry\Policy\CallableRetryPolicy;
+use Retry\RetryProxy;
+
 trait WorkspaceCredentialsAssertTrait
 {
+    private static $RETRY_FAIL_MESSAGE = 'Credentials should be invalid';
```
Comment (zajca) [inline tests/Backend/WorkspaceCredentialsAssertTrait.php:11]: v traitu nejde const :)

**Comment by zajca** [comment]
Testoval sem to tak, že sem si zakomentoval `SynapseWorkspaceProvisioningBackend::resetWorkspacePassword` a díval sa jestli to fakt dělá retry, protože sem synapse nedonutil aby to neprošlo jinak. Zkusil sem aj snflk a normálně to funguje.

**Comment by fejjo** [comment]
👁️

**Review by fejjo — APPROVED** [review]
👍

**Pull request merged**